src/types/coefficient.rs

```diff
@@ -39,8 +39,9 @@ impl Coefficient {
         &self.magnitude
     }
 
-    pub(crate) fn digits(&self) -> u64 {
-        self.magnitude.digits()
+    /// Returns the number of digits in the non-scaled integer representation of the coefficient.
```

Comment:
Scaled vs non-scaled is a `Decimal` concept, so I think you should leave it out of this `Coefficient` doc comment.
```suggestion
    /// Returns the number of digits in the base-10 representation of the coefficient
```